shell/platform/android/io/flutter/plugin/mouse/MouseCursorPlugin.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.mouse;
+
+import android.content.Context;
+import android.view.PointerIcon;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.MouseCursorChannel;
+import java.util.HashMap;
+
+/** A mandatory plugin that handles mouse cursor requests. */
+public class MouseCursorPlugin {
+  @NonNull private final MouseCursorViewDelegate mView;
+  @NonNull private final MouseCursorChannel mouseCursorChannel;
+  @NonNull private final Context context;
+
+  public MouseCursorPlugin(
+      @NonNull MouseCursorViewDelegate view,
+      @NonNull MouseCursorChannel mouseCursorChannel,
+      @NonNull Context context) {
+    mView = view;
+    this.context = context;
+
+    this.mouseCursorChannel = mouseCursorChannel;
+    mouseCursorChannel.setMethodHandler(
+      new MouseCursorChannel.MouseCursorMethodHandler() {
+        @Override
+        public void activateSystemCursor(@NonNull String kind) {
+          mView.setPointerIcon(resolveSystemCursor(kind));
+        }
+      });
+  }
+
+  /**
+   * Return a pointer icon object for a system cursor.
+   *
+   * This method guarantees to return a non-null object.
+   */
+  private PointerIcon resolveSystemCursor(@NonNull String kind) {
+    if (MouseCursorPlugin.systemCursorConstants == null) {
+      MouseCursorPlugin.systemCursorConstants = new HashMap<String, Integer>() {{
+        put("none", new Integer(PointerIcon.TYPE_NULL));
+        //  "basic": default
+        put("click", new Integer(PointerIcon.TYPE_HAND));
+        put("text", new Integer(PointerIcon.TYPE_TEXT));
+        //  "forbidden": default
+        put("grab", new Integer(PointerIcon.TYPE_GRAB));
+        put("grabbing", new Integer(PointerIcon.TYPE_GRABBING));
+      }};
+    }
+
+    final Integer cursorConstantObj = MouseCursorPlugin.systemCursorConstants.get(kind);
```

Comment:
Java autoboxing ints to Integer and unboxing should be able to handle this without the temporary Integer object